msprime/mutations.py

```diff
@@ -713,8 +760,13 @@ class InfiniteSites(MatrixMutationModel):
     # of an earlier design. The class should be formally deprecated and removed at
     # some point.
     def __init__(self, alphabet=BINARY):
+        warnings.warn(
+            "The mutation model class InfiniteSites was deprecated in 1.0."
+            " Use an appropriate alternative such as BinaryMutationModel"
+            " or JC69MutationModel"
```

Comment:
InfiniteAlleles should be added to the list of suggestions. It's probably the closest to InfiniteSites, really, since it (almost) shares the property that you can see all the mutations (well, except when more than one occurs on the same branch).

Reply:
I think this should be a `FutureWarning`, no?

Reply:
https://github.com/tskit-dev/msprime/pull/1176/commits/ca5349e51f979bbffc5458d8a1d04850ec71e8f0